openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -775,10 +775,11 @@ Content-Length: 0
 The Credential Issuer provides a nonce value in the HTTP response with a 2xx status code and the following parameters included as top-level members in the message body of the HTTP response using the application/json media type:
 
 * `c_nonce`: REQUIRED. String containing a nonce to be used when creating a proof of possession of the key proof (see (#credential-request)).
-* `c_nonce_expires_in`: OPTIONAL. Number denoting the lifetime in seconds of the `c_nonce`. This value serves only as a hint to the Client, indicating how long the Credential Issuer is likely to accept the `c_nonce` as valid.
 
 Due to the temporal and contextually sensitive nature of the `c_nonce` value, the Credential Issuer MUST make the response uncacheable by adding a `Cache-Control` header field including the value `no-store`.
 
+A wallet SHOULD assume that the returned `c_nonce` value remains valid and continue using it in credential requests (see [#credential-request]) until the credential endpoint returns an `invalid_nonce` error response.
```

Comment:
IMO this is unneeded and even suggests behavior that might be problematic with respect to efficient (from the client/wallet accessing its local or remote security thingy) processing. We should, again IMO, suggest getting a new nonce prior to each credential endpoint invocation or just say nothing. 
```suggestion
```

Reply:
> We should, again IMO, suggest getting a new nonce prior to each credential endpoint invocation or just say nothing.

:+1: 

Indeed, it would be much cleaner for the wallets to obtain a fresh `c_nonce` before placing a credential request.

I think a recommendation should be included to the specification, just to avoid issuers & wallets to treat `c_nonce` as an ad-hoc refreshable  `DPOP-Nonce`

Reply:
discussed in a WG. rough consensus to remove this paragraph. and ensure that there is enough guidance around returning fresh nonces (seems to be already the case, but @tplooker to double check)
jfyi: DPoP language https://datatracker.ietf.org/doc/html/rfc9449#name-authorization-server-provid

Reply:
I think this sentence came out of my fear that wallets would interpret the spec as "just fetch a fresh nonce each time" once we removed `c_nonce_expires_in`.  But it seems several people actually think wallets potentially doing that is an acceptable outcome. I'm good with dropping this sentence.